fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -440,4 +439,23 @@ public CompletableFuture<IndexOperationResult> performOperation(@Nonnull IndexOp
         return CompletableFuture.completedFuture(new IndexOperationResult() {
         });
     }
+
+    /**
+     * Simple throttling mechanism for log messages.
+     * Since the index writer may see many of these errors in quick succession, limit the number of log messages by ensuring
+     * we only log every so often. This is a very simple mechanism that can be generalized if needed by holding the types of
+     * messages and the required timeouts.
+     * @param format the message format for the log
+     * @param arguments teh message arguments
+     */
+    private void logSerializationError(String format, Object ... arguments) {
+        if (LOG.isWarnEnabled()) {
+            long now = System.currentTimeMillis();
+            if ((now - lastSerializerLog) > SERIALIZER_LOG_DELAY) {
+                LOG.warn(format, arguments);
+                // Not thread safe but OK as we may only log an extra message
```

Comment:
I think for this to be acceptably thread safe you have to mark the field as `volatile`, otherwise weird things happen.
https://docs.oracle.com/javase/tutorial/essential/concurrency/atomic.html

Reply:
Done.